SU2_CFD/src/drivers/CDiscAdjSinglezoneDriver.cpp

```diff
@@ -152,18 +152,6 @@ void CDiscAdjSinglezoneDriver::Run() {
     if(!config->GetTime_Domain() && (MainVariables == FLOW_CONS_VARS))
       config->SetExtIter(Adjoint_Iter);
 
-    /*--- Secondary sensitivities must be computed with a certain frequency. ---*/
-    /*--- It is also done at the beginning so all memory gets allocated.     ---*/
-    if ((Adjoint_Iter % config->GetWrt_Sol_Freq() == 0) && (SecondaryVariables != NONE)){
```

Comment:
This `SecondaryRecording` is now removed entirely, as it introduces a bug for the unsteady adjoint. The Volume Mesh Sensitivity setting is a `+=` in each physical timestep for unsteady cases and `SecondaryRecording` is also called in `Postprocess() `right after the `Run()`. I.e. if it is called multiple times in one physical timestep the sensitivity adds up. With this code modification `SecondaryRecording` gets only called once per timestep in Postprocess.

For steady cases the implications are: The memory for the secondary recording gets allocated after the 'flow adjoint iteration' ... so it could be really annoying if the simulation crashes due to lack of memory after waiting for convergence. More importantly: You cannot use an intermediate restart file for e.g. DOT_AD as it is does not hold any/correct sensitivities. So it might be better to only disable this conditional for unsteady computations as I see the advantage of having this for steady computations. What do you think @rsanfer

Reply:
I honestly prefer it that way as it is much faster for steady-state computations in moving meshes (a lot less unnecessary recordings). In fact that was my first implementation, but then I added this SecondaryRecording after discussion with @talbring, because of the drawbacks you mention. No 100% what's the best approach...

Reply:
My only concern is the memory thing, because usually the recording with coordinates as inputs takes more memory than the one with the solution variables. I am not so worried about the intermediate restart files, since you can just do a restart to get the correct sensitivity values. 

Lets do it without the SecondaryRecording. If it really leads to problems, we can always change it back.

Reply:
Ok, sounds good to me.